src/Antiforgery/src/AntiforgeryMiddleware.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Antiforgery.Infrastructure;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Antiforgery.Internal;
+
+internal sealed class AntiforgeryMiddleware(IAntiforgery antiforgery, RequestDelegate next)
+{
+    private readonly RequestDelegate _next = next;
+    private readonly IAntiforgery _antiforgery = antiforgery;
+
+    private const string AntiforgeryMiddlewareWithEndpointInvokedKey = "__AntiforgeryMiddlewareWithEndpointInvoked";
+    private static readonly object AntiforgeryMiddlewareWithEndpointInvokedValue = new object();
+
+    public async Task Invoke(HttpContext context)
+    {
+        var endpoint = context.GetEndpoint();
+
+        if (endpoint is not null)
+        {
+            context.Items[AntiforgeryMiddlewareWithEndpointInvokedKey] = AntiforgeryMiddlewareWithEndpointInvokedValue;
+        }
+
+        var method = context.Request.Method;
+        if (HttpMethods.IsGet(method) ||
+            HttpMethods.IsHead(method) ||
+            HttpMethods.IsTrace(method) ||
+            HttpMethods.IsOptions(method))
+        {
+            await _next(context);
```

Comment:
Maybe try to avoid async state machine in this case? Similar to:

https://github.com/dotnet/aspnetcore/blob/82e1738c4525bb74d11dbbe856a771b52b951ef6/src/Middleware/OutputCaching/src/OutputCacheMiddleware.cs#L85-L90